File: rails/app/models/core_node.rb
class CoreNode < ActiveRecord::Base
  has_paper_trail

  include Containable
  contains :base_station_units

  belongs_to :status
  belongs_to :city
  belongs_to :location

  def base_station_sectors
    base_station_units.map(&:base_station_sectors).flatten
  end

  def client_links
    base_station_sectors.map(&:client_links).flatten
  end
end

File: ember/app/models/core-node.js
import DS from 'ember-data';

export default DS.Model.extend({
  name:             DS.attr('string'),
  status:           DS.belongsTo('status', { async: true }),
  city:             DS.belongsTo('city', { async: true }),
  baseStationUnits: DS.hasMany('base-station-unit', {async: true}),
  location:         DS.belongsTo('location', {async: true}),

  clientLinks: function() {
    // This returns a flattened list of clientlinks for this core-node
    var clientLinks = [],
        baseStationUnits = this.get('baseStationUnits');

    baseStationUnits.forEach( function(baseStationUnit){
      baseStationUnit.get('baseStationSectors').forEach( function(baseStationSector) {
        baseStationSector.get('clientLinks').forEach(function (clientLink){
          clientLinks.push(clientLink);
        });
      });
    });

    return clientLinks;
  }.property('baseStationUnits'),

  latitude: function() {
    return this.get('location.geometry.latitude');
  }.property('location'),

  longitude: function() {
    return this.get('location.geometry.longitude');
  }.property('location')
});
